pandas/core/apply.py

```diff
@@ -294,7 +299,7 @@ def agg_list_like(self) -> DataFrame | Series:
         """
         from pandas.core.reshape.concat import concat
 
-        obj = self.obj
+        obj = self._maybe_filter_numeric_only()
```

Comment:
This will cause different behaviors for list and non-list arguments that contain a callable. E.g.

```
df.agg(lambda x: x.sum(), numeric_only=True))
df.agg([lambda x: x.sum()], numeric_only=True)
```

Currently in this PR, I believe the first line will pass `numeric_only` to the supplied UDF (and this fails as the UDF doesn't take a numeric_only argument). The second line will subset the DataFrame to only numeric columns and succeed. These should have the same behavior with regards to how the numeric_only argument is treated.

I suggest only using the DataFrame filtered to numeric_only when a given argument in the list is a string; otherwise to use the full DataFrame and pass the numeric_only kwarg to the supplied UDF.